File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.cc
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
	int ary[3];
	for (;;) {
		scanf("%d%d%d", &ary[0], &ary[1], &ary[2]);
		sort(ary, ary + 3);
		if (ary[0] == ary[1] && ary[1] == ary[2]) {
			if (ary[0] == 0) break;
			printf("Equilateral\n");
		}
		else {
			if (ary[2] >= ary[0] + ary[1]) printf("Invalid\n");
			else if (ary[0] == ary[1] || ary[0] == ary[2] || ary[1] == ary[2])
				printf("Isosceles\n");
			else printf("Scalene\n");
		}		
	}
	return 0;
}

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
while True:
  a, b, c = map(int, input().split())

  a = [a, b, c]
  if a[0] == 0 and a[1] == 0 and a[2] == 0:
    break
  
  a.sort()
  if a[0] + a[1] > a[2]:
    if a[0] == a[1] == a[2]:
      print('Equilateral')
    elif a[0] != a[1] != a[2]:
      print('Scalene')
    else:
      print('Isosceles')
  else:
    print('Invalid')